### DISCUSSION TITLE
[Feature] Store pictures and videos locally in iOS app

### DISCUSSION THREAD

**Discussion by fjulian** [immich-app/immich#10840] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

First of all, thank you for this great tool. It offers a very nice way to sort and enjoy photos and videos!

It would be great to have the option to store photos and videos offline on a mobile device (iOS) for when the server is not available. My use case is that when I am on the go with a weak or mobile data connection, I don't want to stream videos. However, if I know beforehand that I will want to show a video to a friend, it would be great if I could make it available offline in the app while I still have a good connection to the server. Another use case would be to access pictures marked as "available offline" even when the server is not available, e.g. because it doesn't run 24/7 because it is based on a NAS in a living space.
Right now, it's already possible to download pictures into the native iOS photo library. However, this creates a copy as far as I can tell. It is cumbersome to make sure that the exported media is not imported to immich again, leading to duplicates. Furthermore, having the downloaded media in the native photo library removes them from their context (albums, time line etc.).
Instead, I would imagine the option to mark individual pictures/videos or an album as "available offline", allowing to display/play them in the immich app even when there is no internet connection, or when the server is down. This would be analogous to downloading music in the Spotify app. The music is not stored outside of the app, but can still be played through the app or as part of a playlist when the device has no internet connection.

Sorry if this already exists and I overlooked it, but I didn't find a way to achieve this in the current app version.

I would also happy to contribute to implementing this feature. Since I'm very new here, any pointers on how to start implementing this would be greatly appreciated.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
#4466/#5993/#7627
